vertical-pod-autoscaler/pkg/recommender/input/cluster_feeder.go

```diff
@@ -358,6 +362,9 @@ func (feeder *clusterStateFeeder) LoadPods() {
 		}
 	}
 	for _, pod := range pods {
+		if feeder.memorySaveMode && !feeder.matchesVPA(pod) {
```

Comment:
I could change it so that when the memory saver mode is on then only pod with matching labels are requested. This would eliminate the need to match the labels but would increase the number of calls to the API server because each set of pods matching the labels would have to be fetched one at a time. The same for container metrics.